plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -108,7 +108,8 @@ func isVolumeConflict(volume api.Volume, pod *api.Pod) bool {
 		if volume.RBD != nil && existingVolume.RBD != nil {
 			mon, pool, image := volume.RBD.CephMonitors, volume.RBD.RBDPool, volume.RBD.RBDImage
 			emon, epool, eimage := existingVolume.RBD.CephMonitors, existingVolume.RBD.RBDPool, existingVolume.RBD.RBDImage
-			if haveSame(mon, emon) && pool == epool && image == eimage {
+			// same RBD image mounted by multiple Pods conflicts unless all Pods mount the image read-only
+			if haveSame(mon, emon) && pool == epool && image == eimage && !(volume.RBD.ReadOnly && existingVolume.RBD.ReadOnly) {
```

Comment:
so the other checks don't matter for read only? if so, can we just check the rw mode on the RDB instead? or please add a comment about why the same mon/emon/pool won't work for read write and why it isn't an issue for read only (I'm pretty new to ceph).